include/future_queue.hpp

```diff
@@ -356,6 +356,8 @@ namespace cppext {
        * called. */
       virtual ~shared_state_base() {}
 
+      inline void free(); // added during Feb 2023 Focus.
```

Comment:
This comment is unnecessary. You can always look it up with git blame. Comments like these should not be in the code.
The function itself does not have a documentation comment. This should say that free decreases the reference count and calls ``delete()`` on the object itself.  The calling code must not use the pointer to the object any more after calling ``free()``.

Following line 349 (class description) the comment should be extended to describe that instances of the class must only be created dynamically with ```new()```, but the creator must not call ```delete()``` but ```shared_state_base```'s ``free()`` function. After this, they must not use the pointer to the object any more because it deletes itself. (I think the last line can be repeated from the ```free()``` function comment because it's important and unusual.)

Reply:
I removed the comment pointed out on line 359 as suggested. Comments added as suggested, though I state that the shared_state_base object must no longer be used. Additionally, I'm adding an explanatory comment above the [if(when_any_notification...](https://github.com/ChimeraTK/cppext/blob/e758f9a3aaa7158602f09421ddc34ed139139c9a/include/future_queue.hpp#L504-L512)   
`      if(executeDelete) {
        // Now that all potential internal references have been cleared the
        // reference count must be 0
        assert(reference_count == 0);
        if(when_any_notification.load().notifyerQueue) {
          when_any_notification.load().notifyerQueue->free();
        }
        delete this;
      } // end if executeDelete`